app/plugin.go

```diff
@@ -594,17 +595,14 @@ func (a *App) getPluginsFromFolder() (map[string]*pluginSignaturePath, *model.Ap
 		return nil, model.NewAppError("getPluginsFromDir", "app.plugin.sync.list_filestore.app_error", nil, appErr.Error(), http.StatusInternalServerError)
 	}
 	pluginSignaturePathMap := make(map[string]*pluginSignaturePath)
-	for _, path := range fileStorePaths {
-		if strings.HasSuffix(path, ".tar.gz") {
-			id := strings.TrimSuffix(filepath.Base(path), ".tar.gz")
-			helper := &pluginSignaturePath{
-				pluginId:      id,
-				path:          path,
-				signaturePath: "",
-			}
-			pluginSignaturePathMap[id] = helper
-		}
+	populatePluginSignatureMap(pluginSignaturePathMap, fileStorePaths, ".tar.gz")
+
+	cannaryTag := *a.Config().PluginSettings.CanaryTag
+	if cannaryTag != "" {
+		canarySuffix := fmt.Sprintf(".tar.gz%s", cannaryTag)
```

Comment:
Could we append the canary tag as `.tar.gz+%s`? Assuming that won't cause weird filename problems, I think it will be clear to see files named `plugin_id.tar.gz+daily` than stumbling over `plugin_id.tar.gzdaily`.